tui/media_config.go

```diff
@@ -0,0 +1,427 @@
+			if child.MountPoint == "/" {
+				child.Type = storage.BlockDeviceTypeCrypt
+			}
+		}
+	}
+
+	// TODO start using new API page.GotoPage() when finished merging
+	// disk pages
+	page.tui.gotoPage(TuiPageMenu, page)
+
+	return false
+}
+
+// Activate updates the UI elements with the most current list of block devices
+func (page *MediaConfigPage) Activate() {
+
+	page.confirmBtn.SetEnabled(false)
+
+	if len(page.safeTargets) == 0 && len(page.destructiveTargets) == 0 {
+		if err := page.buildMediaLists(); err != nil {
+			page.Panic(err)
+		}
+	}
+
+	if page.isSafeSelected {
+		page.activated = page.safeRadio
+	} else if page.isDestructiveSelected {
+		page.activated = page.destructiveRadio
+	} else {
+		page.activated = page.cancelBtn
+	}
+
+	clui.RefreshScreen()
+}
+
+func (page *MediaConfigPage) setConfirmButton() {
+	if page.labelWarning.Title() == "" {
+		page.confirmBtn.SetEnabled(true)
+	} else {
+		page.confirmBtn.SetEnabled(false)
+	}
+}
+
+// The disk page gives the user the option so select how to set the storage device,
+// if to manually configure it or a guided standard partition schema
+func newMediaConfigPage(tui *Tui) (Page, error) {
+	page := &MediaConfigPage{
+		BasePage: BasePage{
+			// Tag this Page as required to be complete for the Install to proceed
+			required: true,
+		},
+	}
+	page.setupMenu(tui, TuiPageMediaConfig, mediaConfigMenuTitle, CancelButton|ConfirmButton, TuiPageMenu)
+
+	// Top label for the page
+	lbl := clui.CreateLabel(page.content, 2, 1, mediaConfigTitle, clui.Fixed)
+	lbl.SetPaddings(0, 1)
+
+	contentFrame := clui.CreateFrame(page.content, AutoSize, AutoSize, BorderNone, Fixed)
+	contentFrame.SetPack(clui.Vertical)
+
+	// Installation Type Radio button selection
+	radioButtonFrame := clui.CreateFrame(contentFrame, AutoSize, AutoSize, BorderNone, Fixed)
+	radioButtonFrame.SetPack(clui.Vertical)
+	radioButtonFrame.SetPaddings(2, 1)
+
+	page.group = clui.CreateRadioGroup()
+	radioLabel := fmt.Sprintf("Safe Installation: Using Free Space")
+	page.safeRadio = clui.CreateRadio(radioButtonFrame, 50, radioLabel, AutoSize)
+	page.safeRadio.SetPack(clui.Horizontal)
+	page.group.AddItem(page.safeRadio)
+	page.safeRadio.OnChange(func(active bool) {
+		if active {
+			page.isDestructiveSelected = false
+		}
+		// Disable the Confirm Button if we toggled
+		if !page.isSafeSelected && active {
+			page.confirmBtn.SetEnabled(false)
+			page.isSafeSelected = true
+
+			page.buildChooserList()
+		}
+
+		if active {
+			if len(page.safeTargets) < 1 {
+				if active {
+					warning := "No safe media found for installation"
+					log.Warning(warning)
+					warning = fmt.Sprintf("Warning: %s", warning)
+					page.labelWarning.SetTitle(warning)
+				}
+			}
+		}
+	})
+
+	// Description of Safe
+	descLabel := clui.CreateLabel(radioButtonFrame, 2, 3, "Uses non-partitioned or gpt partitioned media with free space.\n"+
+		"Install along side existing data or OS if adequate space is available.", Fixed)
```

Comment:
Usage. Change "along side" to one word "alongside"

Reply:
fixed in new force push